sematic/api/endpoints/request_parameters.py

```diff
@@ -339,13 +343,61 @@ def _get_sql_predicates(
         operator = dict(AND=sqlalchemy.and_, OR=sqlalchemy.or_)[operand]
         return operator(
             *[
-                _extract_single_predicate(filter_, column_mapping)
+                _extract_predicate(filter_, column_mapping, model)
                 for filter_ in filters[operand]
             ]
         )
     else:
         filter_ = cast(ColumnPredicate, filters)
-        return sqlalchemy.and_(_extract_single_predicate(filter_, column_mapping))
+        return sqlalchemy.and_(_extract_predicate(filter_, column_mapping, model))
+
+
+def _extract_predicate(
+    filter: ColumnPredicate, column_mapping: ColumnMapping, model: type
+):
+    column_name = list(filter.keys())[0]
+
+    if "." in column_name:
+        return _extract_relationship_predicate(filter, model)
+    else:
+        return _extract_single_predicate(filter, column_mapping)
+
+
+def _extract_relationship_predicate(filter: ColumnPredicate, model: type):
+    filter_key = list(filter.keys())[0]
+    relationship_name, field = filter_key.split(".")
+
+    if not hasattr(model, relationship_name):
+        raise ValueError(f"{relationship_name} is not a field of {model}")
```

Comment:
How can we make it easy for developers / users to discover which relationships are usable on which models?

Please, for example, add docstring entries to each model listing candidates, and direct readers to them from a comment here.

Reply:
added some docs around the Run model.